lang/cs/Org.Apache.REEF.Tang.Tests/Injection/TestInjection.cs

```diff
@@ -304,6 +304,55 @@ public void TestExternalObject()
 
             Assert.IsNotNull(o.ExternalObject is ExternalClass);
         }
+        
+        /// <summary>
+        /// In this test, interface is generic of T. Implementations use different T as type. 
+        /// When doing injection, we need to get type of generic type T so that to get generic interface
+        /// </summary>
+        [TestMethod]
+        public void TestInjectionWithT()
+        {
+            var c = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindImplementation(GenericType<IMyOperator<int>>.Class, GenericType<MyOperatorImpl<int>>.Class)
+                .BindImplementation(GenericType<IMyOperator<int[]>>.Class, GenericType<MyOperatorImpl<int[]>>.Class)
+                .BindImplementation(GenericType<IMyOperator<string>>.Class, GenericType<MyOperatorImpl<string>>.Class)
+                .BindNamedParameter(typeof(MessageType), typeof(int[]).AssemblyQualifiedName)
+                .Build();
+
+            var injector = TangFactory.GetTang().NewInjector(c);
+
+            var msgType = (string)injector.GetNamedInstance(typeof(MessageType));
+            Type t3 = Type.GetType(msgType);
+
+            string str1 = typeof(int).AssemblyQualifiedName;
+            Type t1 = Type.GetType(str1);
+
+            var userType = typeof(IMyOperator<int>);
+            Type genericTypeArgument = null;
+            if (userType.IsGenericType)
+            {
+                var genericTypes = userType.GenericTypeArguments;
+                genericTypeArgument = genericTypes[0];
+            }
+
+            Type myGenericParam2 = typeof(string);
+
+            Type myGenericInterface1 = typeof (IMyOperator<>);
+            Type myBakedGenericInterface1 = myGenericInterface1.MakeGenericType(new Type[] { genericTypeArgument });
+            Type myBakedGenericInterface2 = myGenericInterface1.MakeGenericType(new Type[] { typeof(string) });
+            Type myBakedGenericInterface3 = myGenericInterface1.MakeGenericType(new Type[] { t3 });
+ 
+            //at the time of injection, you must specify which T to use. This is IMyOperator<int> and IMyOperator<strign> would reperent different interface
+            var o1 = TangFactory.GetTang().NewInjector(c).GetInstance(myBakedGenericInterface1);
```

Comment:
Is there a specific need to generate many different injectors here? I'm asking because outside of tests, this situation is likely to occur on the same injector.